test/genesisProtocol.js

```diff
@@ -0,0 +1,393 @@
+  let dao, genesisProtocol, paramsHash, executableTest;
+
+  const createProposal = async function () {
+
+    const result = await genesisProtocol.propose({
+      avatar: dao.avatar.address,
+      numOfChoices: 2,
+      paramsHash: paramsHash,
+      executable: executableTest.address
+    });
+
+    assert.isOk(result);
+    assert.isOk(result.proposalId);
+
+    const stakingToken = await DAOToken.at(await genesisProtocol.stakingToken());
+
+    await stakingToken.approve(accounts[0], web3.toWei(10));
+    await stakingToken.approve(genesisProtocol.address, web3.toWei(10));
+
+    return result.proposalId;
+  };
+
+  const voteProposal = function (proposalId, how) {
+    return genesisProtocol.vote({
+      proposalId: proposalId,
+      vote: how
+    });
+  };
+
+  const stakeProposalVote = function (proposalId, how, amount) {
+    return genesisProtocol.stake({
+      proposalId: proposalId,
+      vote: how,
+      amount: web3.toWei(amount)
+    });
+  };
+
+  beforeEach(async () => {
+
+    dao = await DAO.new({
+      name: "Skynet",
+      tokenName: "Tokens of skynet",
+      tokenSymbol: "SNT",
+      schemes: [
+        { name: "GenesisProtocol" }
+      ],
+      founders: [{
+        address: accounts[0],
+        reputation: web3.toWei(1000),
+        tokens: web3.toWei(1000)
+      }
+      ]
+    });
+
+    const scheme = await dao.getSchemes("GenesisProtocol");
+
+    assert.isOk(scheme);
+    assert.equal(scheme.length, 1);
+    assert.equal(scheme[0].name, "GenesisProtocol");
+
+    genesisProtocol = await GenesisProtocol.at(scheme[0].address);
+
+    assert.isOk(genesisProtocol);
+
+    // all default parameters
+    paramsHash = (await genesisProtocol.setParams({})).result;
+
+    executableTest = await ExecutableTest.deployed();
+  });
+
+
+  it("can call getScoreThresholdParams", async () => {
+    const result = await genesisProtocol.getScoreThresholdParams({
+      avatar: dao.avatar.address,
+    });
+    assert.isOk(result);
+    assert.equal(result.thresholdConstA, 1);
+    assert.equal(result.thresholdConstB, 1);
+  });
+
+  it("can call getShouldBoost", async () => {
+    const proposalId = await createProposal();
+    const result = await genesisProtocol.getShouldBoost({
+      proposalId: proposalId
+    });
+    assert.equal(result, false);
+  });
+
+
+  it("can call getVoteStake", async () => {
+    const proposalId = await createProposal();
+
+    await stakeProposalVote(proposalId, 1, 10);
+
+    const result = await genesisProtocol.getVoteStake({
+      proposalId: proposalId,
+      vote: 1
+    });
+    assert.equal(web3.fromWei(result), 10);
+  });
+
+  it("can call getVotesStatus", async () => {
+    const proposalId = await createProposal();
+    const result = await genesisProtocol.getVotesStatus({
+      proposalId: proposalId
+    });
+    assert.isOk(result);
+    // TODO: currently failing. assert.equal(result.length, 2);  see: https://github.com/daostack/arc/issues/328
```

Comment:
Not sure it is good pointing to github issue at the code ..as when it will change you will need to change your code.

Reply:
I don't understand.  As soon as it changes I intend to change the code, that is,  as soon as the PR is merged and you publish it to NPM.

Reply:
"As soon as it changes I intend to change the code, that is, as soon as the Arc PR with that change is merged and you publish it to NPM."

It will just add another commit to the repo without a real need.

Reply:
I will have to change that line of code in any case, once the related Arc changes are available that affect that line in the test.  Note sure how to avoid that,

Reply:
Why is it failing @dkent600? I also think it is not good to have such github links in the code.

Reply:
@leviadam If you click on the GitHub link you will see why it is failing and what we're planning to do about it.

>  I also think it is not good to have such github links in the code.

I hear you but I'm curious why you think that?

Reply:
Lacking an explanation from you guys on your position, the only thing that would be an improvement, in my eyes, is to create an issue on this, which, since I'm not planning to upgrade to the new Arc package for this PR, is what should be done in any case.  So here it is, and I have removed the controversial commented line of test code:

https://github.com/daostack/arc.js/issues/73